client/src/main/java/com/chattycathy/client/model/User.java

```diff
@@ -18,9 +18,33 @@ public class User {
      * @param scanner takes input for userName
      */
     public User(Scanner scanner) {
+
+        boolean nameNonEmpty;
+        boolean validLength;
+        boolean noSpaces;
+        boolean notServer;
+        boolean validInput;
+
         do {
             log.info("Please enter a username: ");
             userName = scanner.nextLine();
-        } while (userName.isEmpty());
+
+            nameNonEmpty = !userName.isEmpty();
+            validLength = userName.length() <= 30;
+            noSpaces = !userName.contains(" ");
+            notServer = !userName.toLowerCase().contains("server");
+
+            if (!validLength) {
+                log.info("Username must be 30 or less characters");
+            }
+            if (!noSpaces) {
+                log.info("Username cannot contain spaces");
+            }
+            if (!notServer) {
+                log.info("Username cannot be 'server'");
+            }
+
+            validInput = nameNonEmpty && validLength && noSpaces && notServer;
```

Comment:
Logic works well but I think this could be better if it was separated out of the constructor into its own function. This would also mean User wouldn't require any of the boolean variables and line 48 could be simplified by using the result from the function (I will leave another comment for this). I've provided an example of what I'm thinking below but feel free to adjust as you see fit and happy to discuss if you have any questions.

```suggestion
            private boolean isValidUsername(String username) {
                if (userName.isEmpty()) {
                     log.info("...");
                     return false;
                }
                if (userName.length() > 30 || userName.length() < 2) {
                    log.info("...");
                    return false;
                }
                
                // repeat for rest of validation cases
                
                return true;
            }
```

Reply:
I don't think only getting 1 error is an issue, I think its worth it for the readability and reduced checks. for example now the user would interact like so:

"Really long string with spaces!!!"
*Warned about length*
"Shorter String"
*Warned about spaces*
"Shorter_String"
*Valid*

it takes slightly longer but i think its worth it. happy to take feedback if i should add back in the variables to print all warnings at once.